graylog2-server/src/main/java/org/graylog2/cluster/leader/AutomaticLeaderElectionService.java

```diff
@@ -93,7 +98,8 @@ private void runIteration() {
         final boolean wasLeader = isLeader;
 
         try {
-            isLeader = lockService.lock(RESOURCE_NAME).isPresent();
+            optionalLock = lockService.lock(RESOURCE_NAME, null);
+            isLeader = optionalLock.isPresent();
```

Comment:
Are the changes to this class needed? It would be great if we don't have to store the lock as a field if it's unnecessary. I think we don't have to modify this class at all if we add back the `lock(resource)` and `unlock(resource)` methods as suggested in the `LockService` comments. :slightly_smiling_face:

Reply:
@mpfz0r My concern is that we add a mutable field to the class even though it's not needed for the locking API. I think it's good practice to minimize mutable fields in a class.

What is the improvement you see with storing the lock? :slightly_smiling_face:

Reply:
Feels more natural: `here is your lock object` ... `thanks, now have it back please` :-)

Reply:
Okay, fair enough. :slightly_smiling_face: